Orange/widgets/credentials.py

```diff
@@ -0,0 +1,32 @@
+import keyring
+
+SERVICE_NAME = 'Orange3 - {}'
+
+
+class CredentialManager:
+    def __init__(self, username):
+        """
+        Class for storing passwords in the system keyring service.
+
+        Args:
+            username: username used for storing a matching password.
+
+        Examples:
+        >>> cm = CredentialManager('Foo')
+        >>> cm.key = 'Bar'      # store password
+        >>> cm.key              # retrieve password
+        'Bar'
+        """
+        self.username = username
+        self.service_name = SERVICE_NAME.format(self.username)
+
+    @property
+    def key(self):
+        return keyring.get_password(self.service_name, self.username)
+
+    @key.setter
+    def key(self, value):
+        keyring.set_password(self.service_name, self.username, value)
+
+    def delete_password(self):
+        keyring.delete_password(self.service_name, self.username)
```

Comment:
Can these methods be replaced with `__setattr__`, `__getattr__`, and `__delattr__` so the supported interface becomes more like a general secure key-value store?